ecs-cli/integ/resources/ecs_cli_integ_template.json

```diff
@@ -0,0 +1,241 @@
+        },
+        "EcsInstanceRole": {
+            "Type": "AWS::IAM::Role",
+            "Properties": {
+                "RoleName": "ecs-cli-integ-instance-role",
+                "AssumeRolePolicyDocument": {
+                    "Version": "2012-10-17",
+                    "Statement": [
+                        {
+                            "Effect": "Allow",
+                            "Principal": {
+                                "Service": [
+                                    "Fn::If": [
+                                        "IsCNRegion",
+                                        "ec2.amazonaws.com.cn",
+                                        "ec2.amazonaws.com"
+                                    ]
+                                ]
+                            },
+                            "Action": [
+                                "sts:AssumeRole"
+                            ]
+                        }
+                    ] 
+                },
+                "ManagedPolicyArns": [ 
+                    "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role" 
+                ]
+            }
+        },
+        "EcsInstanceProfile": {
+            "Type": "AWS::IAM::InstanceProfile",
+            "Properties": {
+                "Roles": [{ "Ref": "EcsInstanceRole" }]
+            }
+        },
+        "EcsInstanceLc": {
+            "Type": "AWS::AutoScaling::LaunchConfiguration",
+            "Properties": {
+                "AssociatePublicIpAddress": true,
+                "ImageId": {
+                    "Ref": "EcsAmiId"
+                },
+                "InstanceType": "m5.large",
+                "IamInstanceProfile": {
+                    "Ref": "EcsInstanceProfile"
+                },
+                "SecurityGroups": [{
+                    "Ref": "EcsSecurityGroup"
+                }],
+                "UserData": {
+                    "Fn::Base64": {
+                        "Fn::Join": ["", [
+                            "#!/bin/bash\n",
+                            "echo ECS_CLUSTER=ecs-cli-integ >> /etc/ecs/ecs.config"
+                        ]]
+                    }
+                }
+            }
+        },
+        "EcsInstanceAsg": {
+            "Type": "AWS::AutoScaling::AutoScalingGroup",
+            "Properties": {
+                "VPCZoneIdentifier": [{ "Ref": "PubSubnetAz1"}, {"Ref": "PubSubnetAz2"}],
+                "LaunchConfigurationName": {
+                    "Ref": "EcsInstanceLc"
+                },
+                "MinSize": "0",
+                "MaxSize": "10",
+                "DesiredCapacity": "3",
+                "Tags": [
+                    {
+                      "Key": "Name",
+                      "Value": "ECS CLI Integ cluster",
+                      "PropagateAtLaunch": true
+                    }
+                ]
+            }
+        },
+        "EcsCluster": {
+            "Type" : "AWS::ECS::Cluster",
```

Comment:
Question: How come we create the ECS Cluster separately using the API in the ECS CLI? I remember once asking this and it was suggested that there was some dependency in the template that meant it had to exist prior. If that's the case, what's different about this template versus the ECS CLI cluster up one?